virttest/qemu_devices/qdevices.py

```diff
@@ -1150,9 +1150,23 @@ def hotplug_hmp(self):
             out = "device_add %s" % _convert_args(self.params)
         return out
 
+    def _refresh_hotplug_props(self, params):
+        """
+        Refresh hotplug optional props as per params.
+
+        :return: A dict containing hotplug optional props.
+        """
+        return params
+
     def hotplug_qmp(self):
         """:return: the hotplug monitor command"""
-        return "device_add", self.params
+        params = self.params.copy()
+        params = self._refresh_hotplug_props(params)
+        for option in ["node", "requested-size"]:
+            value = params.get(option)
+            if value:
+                params[option] = int(value)
```

Comment:
The logic with L1165~L1168 belong to the `refresh_hotplug_procs` function.  And for node, it already be covered in the `Dimm` class.